### DISCUSSION TITLE
IndexError: list index out of range

### DISCUSSION THREAD

**Discussion by omkarade** [coqui-ai/TTS#1729] [General Q&A] [answered]
![Screenshot (341)](https://user-images.githubusercontent.com/90675376/178745234-32968459-90ad-4e69-87ad-5917183bcd23.png)

**Comment by Konstant1982** [comment]
I have the same error, did you find a solution?

**Comment by mepc36** [comment]
Same error, any solution?

**Comment by mepc36** [comment, marked as answer]
@Konstant1982 Thanks to the Tacotron forums [here](https://github.com/NVIDIA/tacotron2/issues/221) I was able to get past this error by removing the final blank line in my metadata.csv file. Weird, because the train command still worked when I ran it with LJSpeech's own metadata.csv file, which has its own empty blank last line too...weird!

**Comment by gnmarten** [comment]
in my case, this exact error was due to an inconsistency in the metadata.csv file (rows having fewer than 2 columns). 

You can run a sanity check like this to detect the faulty rows (and correct them manually)

```
filename = "metadata.csv"
with open(filename, "r", encoding="utf-8") as f:
    row_num = 0
    for line in f:
        cols = line.strip().split("|")
        if len(cols) < 2:
            print(f"Row {row_num + 1} has fewer than 2 columns: {line.strip()}")
        row_num += 1
```

**Comment by VArdulov** [comment]
So digging into [the relevant code](https://github.com/coqui-ai/TTS/blob/30109af2a011e32b2a6ec73680186fd261c3c61e/TTS/tts/datasets/formatters.py#L152), it seems like the issue is coming from the format of the metadata file and the formatter that you choose. It looks like you're using the `ljspeech` formatter which I think is set as the default in many of the recipes for training/tuning. And if you are like me and you followed [the guide to preparing the dataset in the documentation](https://tts.readthedocs.io/en/latest/formatting_your_dataset.html) for TTS training/tuning, then you'll notice that either there is an error in the ljspeech formatter code that grabs `cols[2]` instead of `cols[1]` which is throwing the `Index Out of Bounds` error or the recommended format in the data preparation is not the LJSpeech format. Looking elsewhere in the guide it looks like the `mozilla` formatter does the same thing and only expects 2 columns. 

I locally installed the library and edited the code to the `ljspeech()` formatter function to pull `cols[1]` instead and this fixed the error and my model was able to train without issue

**Comment by darknight2163** [comment]
> ![Screenshot (341)](https://user-images.githubusercontent.com/90675376/178745234-32968459-90ad-4e69-87ad-5917183bcd23.png)

Follow this 'LJ speech format' for writing on each line of the transcript file: **wav_file_name|transcript|transcript**
eg. : 

LJ001-0002|in being comparatively modern.|in being comparatively modern.
LJ001-0006|And it is worth mention in passing that, as an example of fine typography,|And it is worth mention in passing that, as an example of fine typography,
LJ001-0008|has never been surpassed.|has never been surpassed. 
.
.
.
and so on for all the lines of the file. 
It worked for me so hope this will work for you too👍

**Reply by hrushik98** [reply]
This works! Thanks!

**Comment by siagupta-03** [comment]
![Screenshot (545)](https://github.com/coqui-ai/TTS/assets/112824643/2f9963f3-ebcf-4ad0-9d46-71a19e829c91)
(venv) C:\Users\Sia Gupta\EasyEat\TTS>python train.py
 > Setting up Audio Processor...
 | > sample_rate:22050
 | > resample:False
 | > num_mels:80
 | > log_func:np.log10
 | > min_level_db:-100
 | > frame_shift_ms:None
 | > frame_length_ms:None
 | > ref_level_db:20
 | > fft_size:1024
 | > power:1.5
 | > preemphasis:0.0
 | > griffin_lim_iters:60
 | > signal_norm:True
 | > symmetric_norm:True
 | > mel_fmin:0
 | > mel_fmax:None
 | > pitch_fmin:1.0
 | > pitch_fmax:640.0
 | > spec_gain:20.0
 | > stft_pad_mode:reflect
 | > max_norm:4.0
 | > clip_norm:True
 | > do_trim_silence:True
 | > trim_db:45
 | > do_sound_norm:False
 | > do_amp_to_db_linear:True
 | > do_amp_to_db_mel:True
 | > do_rms_norm:False
 | > db_level:None
 | > stats_path:None
 | > base:10
 | > hop_length:256
 | > win_length:1024
T  File "C:\Users\Sia Gupta\EasyEat\TTS\train.py", line 63, in <module>
    train_samples, eval_samples = load_tts_samples(
                                  ^^^^^^^^^^^^^^^^^
  File "C:\Users\Sia Gupta\EasyEat\TTS\TTS\tts\datasets\__init__.py", line 120, in load_tts_samples
    meta_data_train = formatter(root_path, meta_file_train, ignored_speakers=ignored_speakers)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Sia Gupta\EasyEat\TTS\TTS\tts\datasets\formatters.py", line 201, in ljspeech
    text = cols[2]
           ~~~~^^^
IndexError: list index out of range


I have already cleaned the data with sanity_check and removed the rows but still  geting the same cols error kindly help
